sdk/js-cloud-server/src/cloudClient.ts

```diff
@@ -0,0 +1,233 @@
+import {
+    DevCycleOptions,
+    DVCVariableValue,
+    DVCVariableSet,
+    DVCFeatureSet,
+    DevCycleEvent,
+} from './types'
+import { DVCVariable } from './models/variable'
+import { checkParamDefined } from './utils/paramUtils'
+import { dvcDefaultLogger } from './utils/logger'
+import {
+    DVCPopulatedUser,
+    DevCyclePlatformDetails,
+} from './models/populatedUser'
+import {
+    DVCLogger,
+    getVariableTypeFromValue,
+    VariableTypeAlias,
+} from '@devcycle/types'
+import {
+    getAllFeatures,
+    getAllVariables,
+    getVariable,
+    postTrack,
+    ResponseError,
+} from './request'
+import { DevCycleUser } from './models/user'
+
+const castIncomingUser = (user: DevCycleUser) => {
+    if (!(user instanceof DevCycleUser)) {
+        return new DevCycleUser(user)
+    }
+    return user
+}
+
+/**
+ * Handle response errors from API
+ * Suppress 5xx errors and log them instead while returning defaults to the caller
+ * Throw 4XX errors back to the caller to notify of invalid SDK usage
+ * Special handling of 404 error, which indicates "variable not found"
+ * @param err
+ */
+const throwIfUserError = (err: unknown) => {
+    if (err instanceof ResponseError) {
+        const code = err.status
+
+        // throw the error if it indicates there was user error in this call
+        // (e.g. invalid auth credentials or bad user data)
+        if (code !== 404 && code < 500) {
+            throw new Error(
+                `DevCycle request failed with status ${code}. ${
+                    err.message || ''
+                }`,
+            )
+        }
+
+        // Catch non-4xx errors so we can log them instead
+        return
+    }
+
+    if (err instanceof SyntaxError) {
+        // JSON parse error, log instead of throwing
+        return
+    }
+
+    // if not a ResponseError, throw it
+    throw err
+}
+
+export class DevCycleCloudClient {
```

Comment:
This is the main `DevCycleCloudClient` which was extracted from the NodeJS SDK. The only real change to this class was the addition of the `platformDetails` constructor param to make this agnostic to the platform it is running on.